### PR TITLE
Instrumentation: Start tracing database requests

### PR DESCRIPTION
**Pull request by bergquist**
Start passing `ctx` for all database requests when possible. This does not mean all database requests will be traced, only those who pass the context. 

To start jaeger locally (devenv didn't work for me)
```
docker run \
  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
  -p 5775:5775/udp \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 14268:14268 \
  -p 14250:14250 \
  -p 9411:9411 \
  jaegertracing/all-in-one:1.22
```

ref https://github.com/grafana/grafana/issues/29489

### CODE REVIEW

File: pkg/services/sqlstore/database_wrapper.go [hunk 1]
```diff
@@ -70,11 +73,41 @@ func (h *databaseQueryWrapper) Before(ctx context.Context, query string, args ..
...
+
+	} else {
+		histogram.Observe(elapsed.Seconds())
+	}
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "database query")
+	defer span.Finish()
+
+	span.LogFields(
+		ol.String("query", query),
```
Comment (bergquist) [inline pkg/services/sqlstore/database_wrapper.go:102]: the query does not contain all the params so we won't log any sensitive data.

File: pkg/services/sqlstore/user.go [hunk 2]
```diff
@@ -530,6 +530,7 @@ func GetSignedInUser(ctx context.Context, query *models.GetSignedInUserQuery) er
 		LEFT OUTER JOIN org on org.id = org_user.org_id `
 
 	sess := x.Table("user")
+	sess = sess.Context(ctx)
```
Comment (bergquist) [inline pkg/services/sqlstore/user.go:533]: this could be refactored to use `withinSession` instead.

**Comment by marefr** [comment]
Great. I'll take this for a spin. Is there still a problem enabling database metrics with other databases than mysql - we had pretty serious migration fault with this feature earlier if I remember correctly.

**Comment by bergquist** [comment]
> Is there still a problem enabling database metrics with other databases than MySQL.

Should be fine I think. We tested it on MySQL, postgres and sqlite before merging it behind a feature toggle.

**Review by marefr — APPROVED** [review]
Great :+1: 

Guess I wonder if the success should be in a tag rather in the logs:
image

Guess I don't know best practices around distributed tracing, but I do know that there's a [tag defined in opentracing](https://github.com/opentracing/opentracing-go/blob/3088eee7e4d26010ac9a301e977fd2721dbebcbe/ext/tags.go#L109) which seems for the usecase of putting the DB query in. But maybe a bad idea to put the full query in a tag?

**Comment by marefr** [comment]
Tried to push an update of the jaeger devenv block, but couldn't for some reason so will open a new PR for that

**Comment by bergquist** [comment]
I think tag/logs depends on how the tracing solution store the trace. AFAIK some tracing solutions choose to index the tags and I don't think we should index the tag. This however might be outdated knowledge :)

**Pull request merged**